Userland/Utilities/acpi.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// https://uefi.org/specs/ACPI/6.4/05_ACPI_Software_Programming_Model/ACPI_Software_Programming_Model.html#system-description-table-header
+struct [[gnu::packed]] SDTHeader {
+    char sig[4];
+    u32 length;
+    u8 revision;
+    u8 checksum;
+    char oem_id[6];
+    char oem_table_id[8];
+    u32 oem_revision;
+    u32 creator_id;
+    u32 creator_revision;
+};
+
+static void pkg_length(AK::Detail::ByteBuffer<32> const& data, u32& next_in_block, u32& next_block)
+{
+    u32 pos = next_block;
+
+    outln("");
+    out("data: ");
+    if (pos > 3) {
+        for (auto i = -3; i < 0; i++)
+            out("{:#02x} ", data[pos + i]);
+    } else
+        out("               ");
+    out("  ");
+    for (auto i = 0; i < 5; i++)
+        out("{:#02x} ", data[pos + i]);
+    outln("");
+
+    u32 length = 0;
+    u32 delta = 0;
+    int const size = data[pos + 1] >> 6;
+    if (size == 0) {
+        length = data[pos + 1] & 0x3f;
```

Comment:
Just to make sure (since I can't access the linked specification right now), is the `0x3f` here correct? It doesn't match what the cases for `1 <= size <= 3` are doing.

Reply:
I added a link to the right part of the specification.
yes. 
if the both size bits are zero  the following 6 bits are used for length,
otherwise the 2 top bits of the 6 bits should be zero and the lower 4 bits be used as a part of length